Support for Xamarin.iOS

tpetrina opened the pull request:
Hi,

I need this library for Xamarin iOS and Android so I had to fix some things, here is a short list:

- New `.gitignore` file that removes binaries and packages
- Fixed compile errors
- Added iOS project with linked files

If you need help with creating NuGet packages that can be used in Xamarin projects as well as in .NET, I can help with that too. Currently `ScClient` as available on NuGet is not suitable for Xamarin development.

Cheers
Toni

sacOO7 commented:
Can I ask which packages you have added to .gitignore?

sacOO7 commented:
Also what type of problem is created by current Nuget Package for xamarin developement?

tpetrina commented:
I have ignored `/packages` folder which shouldn't be committed since it is restored on build.

I have added the missing semicolon and fixed the `AsciiEncoding` compile issue too.

I couldn't add `ScClient` to a Xamarin.iOS project, the error I get is
`Could not install package 'ScClient 1.0.0'. You are trying to install this package into a project that targets 'Xamarin.iOS,Version=v1.0', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.`
I am running Xamarin Studio 6.3 and the project is regular Xamarin.iOS project.

sacOO7 commented:
Is current merge code compatible with xamarin-android?

tpetrina commented:
It is, but your NuGet package only works for .NET 4.5 and above. It cannot be used anywhere else. Not sure if you are aware of it, but you need different projects for different runtimes and the final NuGet package consists of a several DLL-s which are then chosen by the target platform.

If you want, I can help you with that. Also, do you plan on using CI like AppVeyor for building NuGet packages? It is free for open source projects and I can help you create such pipeline. It will allow instant publishing of correct NuGet packages to nuget.org once any changes are merged.

sacOO7 commented:
I think you are right about this thing, I will let you know about it. currently busy :p

sacOO7 commented:
Hey I'm ready to accept your pull request, which includes creating dlls for all platforms. Can you create a new pull request, I will merge the code, delete existing package from Nuget and try to upload a new one. Just make sure build supports all platforms mentioned in README file

sacOO7 commented:
Hey, just let me know if you are online.

tpetrina commented:
Hey, I don't think I'll be able to this before Monday. I'll take a look at all the platforms you need me to work on, but I don't think I can cover Windows Phone since I don't want to install that part of VS on my machines.

sacOO7 commented:
Not a problem. You can cover other platforms. Just let me know when you have free time. Then we can finally pack up this thing.

tpetrina commented:
Hi,

I added Android build and I managed to automate NuGet package creation using AppVeyor as it is free for open source projects and is trivial to set up.

I will add remaining build targets later. Please review this and tell me if you approve of it.

Source code has not been touched, I merely built the infrastructure for correct NuGet packaging.

`WebSocket4Net` targets the following platforms

- .Net 2.0
- .Net 3.5
- .Net 4.0
- .Net 4.5
- Mono
- Silverlight 4/5
- WindowsPhone 7.5/8.0
- Xamarin.Android
- Xamarin.iOS (Classic and Unified API)
- Universal Windows Platform
- dotnet core

Which ones do you want to support?

sacOO7 commented:
You can try to support as much of as possible, as listed in readme file :)

sacOO7 commented:
Hey, @tpetrina can you make a new PR? I will take a look at it. It seems the codebase changed a lot since last time

tpetrina commented:
Hey, not really - this was a 4-year-old PR and I am not planning on using this library at all. Just cleaning my PR history.

sacOO7 commented:
That's okay, thanks for your efforts anyways
